src/Aspire.Cli/DotNetCliRunner.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+
+namespace Aspire.Cli;
+
+internal sealed class DotNetCliRunner
+{
+    internal Func<int> GetCurrentProcessId { get; set; } = () => Environment.ProcessId;
+
+    public async Task<int> RunAsync(FileInfo projectFile, string[] args, CancellationToken cancellationToken)
+    {
+        string[] cliArgs = ["run", "--project", projectFile.FullName, "--", ..args];
+        return await ExecuteAsync(cliArgs, cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task<int> InstallTemplateAsync(string packageName, string version, bool force, CancellationToken cancellationToken)
+    {
+        string[] forceArgs = force ? ["--force"] : [];
+        string[] cliArgs = ["new", "install", $"{packageName}::{version}", ..forceArgs];
+        return await ExecuteAsync(cliArgs, cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task<int> NewProjectAsync(string templateName, string name, string outputPath, CancellationToken cancellationToken)
+    {
+        string[] cliArgs = ["new", templateName, "--name", name, "--output", outputPath];
+        return await ExecuteAsync(cliArgs, cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task<int> ExecuteAsync(string[] args, CancellationToken cancellationToken)
+    {
+        var startInfo = new ProcessStartInfo("dotnet")
+        {
+            UseShellExecute = false,
+            CreateNoWindow = true
+        };
+
+        if (args.Length > 0)
+        {
+            startInfo.Arguments = string.Join(" ", args);
```

Comment:
https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.argumentlist?view=net-9.0#system-diagnostics-processstartinfo-argumentlist

> Strings added to the list don't need to be previously escaped. 😮 

```suggestion
            startInfo.ArgumentList = args;
```